packages/broccoli-eyeglass/src/broccoli_sass_compiler.ts

```diff
@@ -644,23 +696,38 @@ export default class BroccoliSassCompiler extends BroccoliPlugin {
       this.addDependency(details.fullSassFilename, path.resolve(details.srcPath, dep));
     });
 
-    let files = Object.keys(outputFiles);
+    let {contents, urls} = outputFiles;
+    let files = Object.keys(contents);
 
-    persistentCacheDebug(
+    this.persistentCacheDebug(
       "cached output files for %s are: %s",
       details.sassFilename,
       files.join(", ")
     );
 
-    files.map(file => {
-      let data = outputFiles[file];
+    for (let file of files) {
+      let data = contents[file];
       let cachedFile = path.join(this.cachePath!, file);
       let outputFile = path.join(this.outputPath, file);
       // populate the output cache for rebuilds
       this.addOutput(details.fullSassFilename, outputFile);
 
       writeDataToFile(cachedFile, outputFile, Buffer.from(data, "base64"));
-    });
+    }
+    let eventPromises: Array<Promise<any>> = [];
+    let allUrls = Object.keys(urls);
+    if (allUrls.length > 0) {
+      this.persistentCacheDebug(
+        "firing 'cached-asset' for each asset url for %s: %s",
+        details.sassFilename,
+        allUrls.join(", ")
+      );
+    }
+    for (let url of allUrls) {
+      let sourceFile = urls[url];
+      eventPromises.push(this.events.emit("cached-asset", sourceFile, url))
+    }
+    return RSVP.all(eventPromises);
```

Comment:
do we want to fast fail, if any of the promises reject? Or should we wait until al the work is done, before proceeding?

Reply:
Both of those behaviors are fine in this situation.